### PR TITLE
Removed redundant isMetricsEnabled in CorfuCompileProxy.

### PR DESCRIPTION
**Pull request by zhangn49**
## Overview
Cleaned up redundant part in CorfuCompileProxy.

Description:
1. Removed redundant boolean isMetricsEnabled, since it refers metricsCollectionEnabled from MetricsUtils then applies it to function of MetricsUtils.
2. Removed unnecessary global variables.
3. Replaced anonymous classes with lambda in MetricsUtils.

Why should this be merged: 
Removed redundant and confused part.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java [hunk 1]
```diff
@@ -154,16 +151,15 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 upcallTargetMap, undoRecordTargetMap,
                 undoTargetMap, resetSet);
 
-        metrics = CorfuRuntime.getDefaultMetrics();
```
Comment (lixinchengdu) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: why MetricRegistry becomes non final?

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java [hunk 2]
```diff
@@ -288,24 +292,14 @@ public static void incConditionalCounter(boolean enabled, @NonNull Counter count
      * return a gauge on direct memory used by netty's PooledByteBufAllocator
      */
     private static Gauge<Long> getNettyPooledDirectMemGauge() {
-        return new Gauge<Long>() {
-            @Override
-            public Long getValue() {
-                return PooledByteBufAllocator.DEFAULT.metric().usedDirectMemory();
-            }
-        };
+        return () -> PooledByteBufAllocator.DEFAULT.metric().usedDirectMemory();
```
Comment (lixinchengdu) [inline runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: Even though the new one is cleaner, the old on is easier to understand.

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java (hunk 1 above)
Comment (zhangn49) [reply runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: I thought local variable does not need to be final. Fixed.

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java (hunk 2 above)
Comment (zhangn49) [reply runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: I agree. I always use IntelliJ to translate lambda to anonymous class when I don't understand. Modified these to keep a unified code style.

**Pull request merged**